File: android/src/main/kotlin/io/openim/flutter_openim_sdk_ffi/listener/OnGroupListener.kt
package io.openim.android.sdk.listener

import io.openim.flutter_openim_sdk_ffi.models.GroupApplicationInfo
import io.openim.flutter_openim_sdk_ffi.models.GroupInfo
import io.openim.flutter_openim_sdk_ffi.models.GroupMembersInfo


/**
 * 群组监听
 */
interface OnGroupListener {
    fun onGroupApplicationAccepted(info: GroupApplicationInfo)
    fun onGroupApplicationAdded(info: GroupApplicationInfo)
    fun onGroupApplicationDeleted(info: GroupApplicationInfo)
    fun onGroupApplicationRejected(info: GroupApplicationInfo)
    fun onGroupInfoChanged(info: GroupInfo)
    fun onGroupMemberAdded(info: GroupMembersInfo)
    fun onGroupMemberDeleted(info: GroupMembersInfo)
    fun onGroupMemberInfoChanged(info: GroupMembersInfo)
    fun onJoinedGroupAdded(info: GroupInfo)
    fun onJoinedGroupDeleted(info: GroupInfo)
}

File: ios/Classes/listener/OnGroupListener.swift


/**
 * 群组监听
 */
protocol OnGroupListener {
    func onGroupApplicationAccepted(_ info: GroupApplicationInfo)
    func onGroupApplicationAdded(_ info: GroupApplicationInfo)
    func onGroupApplicationDeleted(_ info: GroupApplicationInfo)
    func onGroupApplicationRejected(_ info: GroupApplicationInfo)
    func onGroupInfoChanged(_ info: GroupInfo)
    func onGroupMemberAdded(_ info: GroupMembersInfo)
    func onGroupMemberDeleted(_ info: GroupMembersInfo)
    func onGroupMemberInfoChanged(_ info: GroupMembersInfo)
    func onJoinedGroupAdded(_ info: GroupInfo)
    func onJoinedGroupDeleted(_ info: GroupInfo)
}